### DISCUSSION TITLE
How to set the severity level in the allure report

### DISCUSSION THREAD

**Discussion by dissanayake98** [allure-framework/allure2#1512] [Questions & Support]
I am working with the robot framework and using the allure framework to generate the test report. 

When I generate the report for both passed and failed test cases, it displays the severity graph as this, 
![a 2](https://user-images.githubusercontent.com/66441600/145033146-1ccdea42-af80-445b-af85-1fec517ee0b8.png)

When I generate the report for both all passed test cases, it displays the severity graph as this, 
![a3](https://user-images.githubusercontent.com/66441600/145033352-2a69d833-1135-4117-8d13-4e48ea3394bd.PNG)

But, I want to know, how to get the severity graph with various statuses (Failed, Broken, Passed, Skipped, Unknown) like this, 
![a1](https://user-images.githubusercontent.com/66441600/145033439-c0c5409d-6aae-4953-9c20-38eb21a24049.PNG)

I have installed the below command;
> pip install allure-behave

After that, I compiled the code with this command;
> behave -f allure_behave.formatter:AllureFormatter -o NewResults/TestResults ./features

Then I got the below message,
> ConfigError: No steps directory in 'C:\UserUser\MyAutomation\features'

I have no idea about these severity graphs, so I have followed the below links to find the details; [https://docs.qameta.io/allure/#_graphs](https://docs.qameta.io/allure/#_graphs)

[https://docs.qameta.io/allure/#_behave](https://docs.qameta.io/allure/#_behave)

[http://allure.qatools.ru/](http://allure.qatools.ru/)

But I couldn't get any ideas about this. Could you please explain this simply?

Any guidance and advice will be greatly appreciated!

**Comment by sgkiokas** [comment]
Hi @dissanayake98 . In which language have you built your test suite? Because I have achieved the above in Javascript if you need any help.

**Reply by dissanayake98** [reply]
This is my code. I am using the robot framework with python 3.6. If you can help, it will be great. 

` *** Settings ***
Resource     /resource/path/new.robot

Library     DataDriver      file=/resource/path/New.xlsx    encoding=UTF-8

Test Setup          Begin Test
Test Template       Fill the form
Test Teardown       Close Browser

*** Test Cases ***
TC_01_Data_driven_testing_using_excel
   Fill the form

*** Keywords ***

Begin Test
    Open Browser    ${BASE_URL}     gc
    maximize browser window

Fill the form

     [Arguments]     ${Email}  ${Password}  

        input text    ${LOCATOR_Email}    ${Email}
        capture page screenshot  
        input text    ${LOCATOR_Password}    ${Password}
        capture page screenshot  
        Click element  xpath://html/body/app-root/div/form/button
        Sleep  10s


        # Login to the page
        click element  xpath://html/body/app-root/div/app-sidebar/div/ul/li[9]/a/p
        sleep  10s
        capture page screenshot  

        # Select the method 
        click element  xpath://html/body/app-root/div/mat-radio-button/label/span[1]
        Sleep  1s

        ## Enter the date
        input text  ${LOCATOR_Date}      10
        capture page screenshot  
        Sleep  1s

        # Continue button
        Click element  xpath://html/body/app-root/div[/form/button
        Sleep  2s
   `

**Reply by sgkiokas** [reply]
Hmm, you are not using Cucumber, right? In my implementation, I simply tagged the Scenarios I wanted to mark with a different severity with a format such as `@severity:critical` and then put in a custom reporter (maybe it's not the case for you) something like:

```
        labels: {
          epic: [/@epic:(.*)/],
          story: [/@story:(.*)/],
          feature: [/@epic:(.*)/],
          severity: [/@severity:(.*)/],
        },
      }
```

Can you achieve such customisation?

**Reply by dissanayake98** [reply]
Thanks for ur help, but I cannot develop my code like your one. 
Is there any command to get the severity like critical, normal, etc?

**Reply by sgkiokas** [reply]
Have you checked out [this](https://github.com/allure-framework/allure-python/blob/master/allure-pytest/examples/label/severity/severity.rst) and [that](https://stackoverflow.com/questions/55234332/how-to-set-the-severity-level-in-allure-report-using-pytest-bdd)?

**Reply by dissanayake98** [reply]
I tried with these two now. But still, I couldn't do it. Anyhow, I will try to find the method. Thanks again for your help.

**Reply by sujith0980** [reply]
Hello @sgkiokas  I see that you have implemented cucumber with javascript and allure. I have created project with Playwright cucumber using javascript with allure integration. I have added hooks but not adding those hooks to the allure report. Could you please provide a sample repo that includes every file for the setup ? I would really appreciate it if you can help me.

**Comment by Alaaatayran** [comment]
hi
 did you figure it out?
